### PR TITLE
Auth UI Forgot Password - Part 1

### PR DESCRIPTION
**Pull request by louischan-oursky**
This PR rewrites the existing implementation

Changes:

- Able to send reset password link via SMS
- Remove SecureMatch
- Remove success and error redirect
- Remove the ad-hoc form

### CODE REVIEW

File: pkg/auth/dependency/forgotpassword/provider.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+	if err != nil {
+		return
+	}
+
+	if len(prins) <= 0 {
+		err = ErrLoginIDNotFound
+		return
+	}
+
+	for _, prin := range prins {
```
Comment (carmenlau) [inline pkg/auth/dependency/forgotpassword/provider.go:69]: After removing realm, is there any case that we need to handle multiple principals in forgot password?

Comment (louischan-oursky) [reply pkg/auth/dependency/forgotpassword/provider.go:69]: Maybe `username` is configured to allow `@` and a username looks like an email address.
But the loop body does not handle this case correct. It should skip non-email and non-phone login ID.

File: pkg/auth/dependency/forgotpassword/provider.go [hunk 2]
```diff
@@ -84,12 +73,16 @@ func (p *Provider) SendCode(loginID string) (err error) {
 
 		if email {
 			err = p.sendEmail(prin.LoginID, codeStr)
-			return
+			if err != nil {
+				return
+			}
```
Comment (carmenlau) [inline pkg/auth/dependency/forgotpassword/provider.go:78]: I think we should return after sending even there is no error? I think creating and sending multiple code is not expected in forgot password context?

**Comment by louischan-oursky** [comment]
Let me close this and continue in the PR of part 2.

**Pull request closed without merging**